### PR TITLE
IZE-248 IZE-250 symlink is not used and download directory

### PR DESCRIPTION
**Pull request by psihachina**
## Changelog:
- use tf bin path for executing tf
- change download directory

## Test:
[asciicast](https://asciinema.org/a/ITQwH4HpVj9qI9B6k0wWguARb)

### CODE REVIEW

File: internal/commands/terraform/terraform.go [hunk 1]
```diff
@@ -20,6 +20,9 @@ type TerraformOptions struct {
 
 var terraformLongDesc = templates.LongDesc(`
 	Run terraform command via terraform docker container.
+	By default terraform run in docker container.
```
Comment (AutomationD) [inline internal/commands/terraform/terraform.go:23]:
```suggestion
	By default terraform runs in a docker container.
```

File: internal/terraform/local.go [hunk 2]
```diff
@@ -102,15 +108,18 @@ func installVersion(version string, custBinPath *string, mirrorURL *string) erro
...
 
 		if exist {
-			tfswitcher.Install(requestedVersion, *custBinPath, *mirrorURL)
+			installFileVersionPath, err = tfswitcher.Install(requestedVersion, *custBinPath, *mirrorURL)
+			if err != nil {
+				return "", err
+			}
 		} else {
-			return fmt.Errorf("the provided terraform version does not exist")
+			return "", fmt.Errorf("the provided terraform version does not exist")
```
Comment (AutomationD) [inline internal/terraform/local.go:116]:
```suggestion
			return "", fmt.Errorf("provided terraform version does not exist")
```

File: internal/terraform/local.go [hunk 3]
```diff
@@ -102,15 +108,18 @@ func installVersion(version string, custBinPath *string, mirrorURL *string) erro
...
 		}
 
 	} else {
 		tfswitcher.PrintInvalidTFVersion()
-		return fmt.Errorf("args must be a valid terraform version")
+		return "", fmt.Errorf("args must be a valid terraform version")
```
Comment (AutomationD) [inline internal/terraform/local.go:121]:
```suggestion
		return "", fmt.Errorf("argument must be a valid terraform version")
```

File: internal/terraform/local.go [hunk 4]
```diff
@@ -44,13 +45,16 @@ func (l *local) Prepare() error {
 	var (
 		tfpath        = "/usr/local/bin/terraform"
```
Comment (AutomationD) [inline internal/terraform/local.go:46]: @psihachina Can you please clarify, are we still installing terraform into that location? If so, this is not what we want. This path should not be used, and terraform should be executed only from ~/.ize location in my opinion.

File: internal/terraform/tfswitcher/install.go [hunk 5]
```diff
@@ -15,7 +15,7 @@ import (
 const (
 	installFile               = "terraform"
 	versionPrefix             = "terraform_"
-	installPath               = ".terraform.versions"
+	installPath               = ".ize/.terraform.versions/"
```
Comment (AutomationD) [inline internal/terraform/tfswitcher/install.go:18]: I'd say let's consider this:
```suggestion
	installPath               = ".ize/versions/terraform/"
```

File: internal/commands/terraform/terraform.go [hunk 6]
```diff
@@ -20,6 +20,9 @@ type TerraformOptions struct {
...
+
+	At the same time, terraform will be downloaded and launched from ~/.ize/.terraform.versions
```
Comment (AutomationD) [inline internal/commands/terraform/terraform.go:25]:
```suggestion
	At the same time, terraform will be downloaded and launched from ~/.ize/versions/terraform/
```

Comment (AutomationD) [inline internal/commands/terraform/terraform.go:25]: @psihachina can you please why you used "at the same time", and explain the logic? 
Do we still download the binary even if the docker version is used?

File: internal/terraform/tfswitcher/install.go [hunk 7]
```diff
@@ -15,7 +15,7 @@ import (
...
 	recentFile                = "RECENT"
 	defaultBin                = "/usr/local/bin/terraform" //default bin installation dir
```
Comment (AutomationD) [inline internal/terraform/tfswitcher/install.go:20]: See my comment about tfpath and installation above. I don't think we need to use that feature, but instead run the required version directly from the downloaded path or something along those lines.

**Review by AutomationD — CHANGES_REQUESTED** [review]

File: internal/terraform/local.go (hunk 4 above)
Comment (psihachina) [reply internal/terraform/local.go:46]: @AutomationD No, we are not installing terraform to this location, but I will delete this variable in another story about refactoring this package so as not to break anything

File: internal/commands/terraform/terraform.go (hunk 6 above)
Comment (psihachina) [reply internal/commands/terraform/terraform.go]: @AutomationD Please check `update description` commit

**Pull request merged**